pkg/runtime/wfengine/activity.go

```diff
@@ -124,6 +210,68 @@ func (*activityActor) DeactivateActor(ctx context.Context, actorID string) error
 	return nil // nothing to do - this actor maintains no state
 }
 
+func (a *activityActor) loadActivityState(ctx context.Context, actorID string, generation uuid.UUID) (*activityState, error) {
+	// See if the state for this actor is already cached in memory.
+	result, ok := a.statesCache.Load(actorID)
+	if ok {
+		cachedState := result.(*activityState)
```

Comment:
Also, per a separate conversation, should we include the generation in the cache key?

Reply:
Good point about caching a copy vs. a reference. Let me look into that.

I don't want to include the generation in the cache key, though. My intent is that these cache keys be strictly actor IDs. TBD on whether to include generations in cache keys, but I think I want to avoid this as well.

Reply:
I made the change to store/fetch a copy of the cache data instead of a reference. However, I want to leave the generation out of the cache key. There should technically never be two generations of the same activity running twice (the orchestration ensures this) but even if there was, the guarantees of the actor subsystem ensures that we don't have to worry about them running in parallel.

Reply:
Cool, thanks! :+1: